File: Roommate.Api/Entities/Configuration.cs
using Roommate.Api.Models;

namespace Roommate.Api.Entities
{
    public class Configuration :Entity
    {
        public virtual Home Home { get; set; }
        public virtual Duty CleaningKitchen { get; set; }
        public virtual Duty CleaningHall { get; set; }
        public virtual Duty CleaningToilet { get; set; }
        public virtual Duty CleaningBathroom { get; set; }
        public virtual Duty CleaningSalon { get; set; }
        public virtual Duty DishWashing { get; set; }
        public virtual Duty WateringPlants { get; set; }

        public void SetCleaning(Cleaning form)
        {
            CleaningKitchen = form.CleaningKitchen;
            CleaningBathroom = form.CleaningBathroom;
            CleaningHall = form.CleaningHall;
            CleaningSalon = form.CleaningSalon;
            CleaningToilet = form.CleaningToilet;
        }

        public void SetDishWashing(Duty duty)
        {
            DishWashing = duty;
        }

        public void SetWateringPlants(Duty duty)
        {
            WateringPlants = duty;
        }
    }
}

File: Roommate.Web/src/reducers/Home/configuration.js
import { combineReducers } from 'redux';
import cleaningToilet from './cleaningToilet';
import cleaningKitchen from './cleaningKitchen';
import cleaningHall from './cleaningHall';
import cleaningBathroom from './cleaningBathroom';
import cleaningSalon from './cleaningSalon';
import dishWashing from './dishWashing';
import wateringPlants from './wateringPlants';

const cleaning = combineReducers({
    cleaningKitchen,
    cleaningToilet,
    cleaningBathroom,
    cleaningSalon,
    cleaningHall
});

const configurationMessage = (state = null, action) => {
    switch (action.type) {
        case 'SET_CONFIGURATION_SUCCESS':
            return "Ustawienia zostały zapisane.";
        default:
            return null;
    }
};

const tasksMessage = (state = null, action) => {
    switch (action.type) {
        case 'GENERATE_TASKS_SUCCESS':
            return "Grafik został wygenerowany.";
        default:
            return null;
    }
};

const configuration = combineReducers({
    cleaning,
    dishWashing,
    wateringPlants,
    configurationMessage,
    tasksMessage
});

export default configuration;
export const getConfiguration = (state) => state.home.configuration;